### ISSUE TITLE
bug on dump MacOS machO signature

### ISSUE THREAD

**Issue by shuxin** [wbond/asn1crypto#136]
[Chrome.zip](https://github.com/wbond/asn1crypto/files/3061729/Chrome.zip)
when ```cms.ContentInfo.load("chrome.rsa buff").dump()```, It lost two bytes for "EOC"

**Comment by wbond** [comment]
Could you provide some more details, perhaps debug output from asn1crypto and the difference between what should be present?

**Comment by shuxin** [comment]
raw = open("chrome.rsa", "rb").read()
dump = cms.ContentInfo.load(raw).dump()
print raw == dump
--------
False

**Comment by joernheissler** [comment]
```python3
from asn1crypto.core import Integer, Sequence

class Foo(Sequence):
    _fields = [("value", Integer)]

ber_encoded = bytes.fromhex("3080020230390000")
foo = Foo.load(ber_encoded)
print("Native:     ", foo.native)
print("BER encoded:", ber_encoded.hex())
dumped = foo.dump()
print("Dumped:     ", dumped.hex())
der_encoded = foo.dump(True)
print("DER encoded:", der_encoded.hex())
```

Output is:
```
Native:      OrderedDict([('value', 12345)])
BER encoded: 3080020230390000
Dumped:      308002023039
DER encoded: 300402023039
```

I hope I got the BER encoding right with the undefined length.

**Comment by joernheissler** [comment]
```patch
diff --git a/asn1crypto/core.py b/asn1crypto/core.py
index 47013d7..2d0d616 100644
--- a/asn1crypto/core.py
+++ b/asn1crypto/core.py
@@ -617,7 +617,7 @@ class Asn1Value(object):
             self._header = header
             self._trailer = b''
 
-        return self._header + contents
+        return self._header + contents + self._trailer
 
 
 class ValueMap():
```

This appears to fix it and it doesn't break any tests.
But this is pure guess work; so far I didn't bother to understand the inner workings (parsing, dumping, data structures) of asn1crypto.

**Comment by shuxin** [comment]
![DIFF](https://user-images.githubusercontent.com/734629/63111827-244b0380-bfc1-11e9-9fee-c96547ef8749.PNG)
Is it a needed EOC or just a trailer?

**Comment by joernheissler** [comment]
I'm not sure what a `trailer` is in asn1crypto. I *assume* that the trailer **is** the EOC.

**Comment by wbond** [comment]
This should be resolved by c29117fd57deb80fb345cf76cad9d0d48e8bbf17.

If we have the EOC bytes (`_trailer` in asn1crypto parlance), we include them when dumping.

Additionally, if we know the item being dumped is indefinite length (and thus BER-encoded), then we force it to be re-encoded using DER.
